src/batou_ext/oci.py

```diff
@@ -0,0 +1,114 @@
+import hashlib
+import os
+import re
+from typing import Optional
+
+import pkg_resources
+from batou import UpdateNeeded
+from batou.component import Attribute, Component
+from batou.lib.file import File
+from batou.lib.service import Service
+
+import batou_ext.nix
+
+
+@batou_ext.nix.rebuild
+class Container(Component):
+    """
+    A OCI Container component.
+    With this component you can dynamically schedule docker containers to be run on the target host
+
+    Example:
+    ```
+    self += batou_ext.oci.Container("mysql", "8.0")
+    ```
+    """
+
+    entrypoint: Optional[str] = None
+    envfile: Optional[str] = None
+    mounts: dict = {}
+    ports: dict = {}
+    env: dict = {}
+
+    registry_address = Attribute(Optional[str], None)
+    registry_user = Attribute(Optional[str], None)
+    registry_password = Attribute(Optional[str], None)
+
+    _required_params_ = {
+        "image": "mysql",
+    }
+
+    def __init__(
```

Comment:
Interesting. We usually never override `__init__` but do everything in `configure()`

@ctheune What do you think about this?

Reply:
This is highly unusual. I couldn't from the top of my head tell whether this breaks any assumptions about when this code gets executed. At least it does break all conventions of being able to access parent components etc. 

So, yes, this I think this should go into configure as well.